### DISCUSSION TITLE
how to set line break to automatic

### DISCUSSION THREAD

**Discussion by lsq** [lvjr/tabularray#241] [Q&A]
``` latex
\documentclass{ctexart}
%\usepackage{ctex}

\usepackage{enumitem}
\usepackage{tabularray}
\UseTblrLibrary{varwidth} 

\begin{document}

\begin{tblr}{
		hlines,vlines,%rowspec={Q[c,m]*{5}{Q[l,m]}},		
		measure=vbox,stretch=-1,
		column{2} = {5cm,l}
	}
	\emph{物料编码} & \emph{物料名称} & \emph{ERP数量} & \emph{复盘数量} \emph{差异数量} & \emph{差异原因} & \emph{仓库}\\
	6141002100 &
	笔管\textbackslash{}USI127\textbackslash{}PC\textbackslash{}123.9*8.0*6.5\textbackslash{}QFY-6141002100\textbackslash{}A0\textbackslash{}表面喷涂丝印镭雕\textbackslash{}委外 &
	3217 &
	74 &
	-3143 &
	盘亏 &
	原料仓\\
	6141000175 &
	笔尖\textbackslash{}adonit   note\textbackslash{}?2.49*7.7\textbackslash{}QFY-6141000173\textbackslash{}A1\textbackslash{}模内纹理\textbackslash{}伸\kaishu{銘} &
	Pcs &
	1000 &
	0 &
	-1000 &
	盘亏
\end{tblr}
\end{document}
```
When there is more cell content, Chinese hyphenation must be manually hyphenated, how to set it to automatic

**Comment by lvjr** [comment]
**"GitHub Issues" is for reporting bugs, while "[GitHub Discussions](https://github.com/lvjr/tabularray/discussions)" is for asking questions. Therefore I have moved the question here.**

In your example, there is no Chinese hyphenation (──) but only a English hyphenation (-). Therefore I can not understand what you want.

In general you can add some `\allowbreak` commands to tell TeX that these positions are valid breaking points. For example, you can replace some `\textbackslash{}` with `\textbackslash{}\allowbreak` in your code.

**Reply by lsq** [reply]
![图片](https://user-images.githubusercontent.com/1617256/165930799-adba28d1-309b-40b0-9998-b35e695dd5f9.png)
Sorry  for my poor English. The problem is not Chinese hyphenation (──) ,  just like above picture, the breaking points is unreasonable.
Is there an algorithm that can be automated setting breaking points.

**Reply by lvjr** [reply]
In fact, TeX treats
```latex
\textbackslash{}USI127\textbackslash{}PC\textbackslash{}123.9*8.0*6.5\textbackslash{}QFY-6141002100\textbackslash{}A0\textbackslash{}
```` 
as a quite long English word, and can only break lines at the hyphen character (-).

I can not find an automatic setting for changing this behavior. But you can use [`seqsplit`](https://ctan.org/pkg/seqsplit) package and put the long string inside `\seqsplit` command to allow TeX break after any character in the string:
```latex
\seqsplit{\textbackslash{}USI127\textbackslash{}PC\textbackslash{}123.9*8.0*6.5\textbackslash{}QFY-6141002100\textbackslash{}A0\textbackslash{}}
```

![image](https://user-images.githubusercontent.com/4239949/165938505-340b3dbb-7d3a-4542-91ff-1caf8a476be2.png)

**Reply by lsq** [reply]
Very much  appreciate your help. But how to set column{2} using \seqsplit with some options instead using \seqsplit  very time.

**Reply by lvjr** [reply]
You can try `column{2} = {cmd=\seqsplit}`.
